### DISCUSSION TITLE
FR: Allow Control of a Device despite it having communication issues

### DISCUSSION THREAD

**Discussion by udochrist** [SukramJ/aiohomematic#1593] [Ideas]
I don't know if this is something that can only be solved in Homeassistant generally, but I think it also affects the way the homematic integration handles communication errors with devices.

Sometimes opening all shutters using a timer  does not open one of the shutters due to a communication failure.
To prevent getting side-tracked in the discussion of this FR: In the scope of the FR i don't care for the reason why this happens and I don't want to go into discussions what might cause the issue and what I could try to remedy it. 

if this happens I am no longer allowed to control  the shutter in home assistant as it is disabled in the gui.
using the homematic ccu web app I can simply press the up button to control the shutter manually and it is opening each time w/o any issue. The communication error is immediately cleared and the device is usable again even in homeassistant.
since I can do the same using another thirdparty app (@home for homematic) the blocking of control cannot be an issue with the homematic API on the CCU. therefore i would like to ask:

is there a obvious reason why the homeassistant gui/integration does not allow the usage of devices that had a communication failure?
Who would be hurt by allowing an operation of a failed device?

I think its very reasonable to show that a device had/has a failure to operate (and even indicate it much clearer than simply deactivating the gui controls for it), but I don't see why home-assistant needs to block all usage of the device in the gui.

All in all my FR is:

- if a device has a communication failure then indicate that this is the case but let the users control it nevertheless.

**Comment by SukramJ** [comment]
There is already an implemented solution for a longer time.

See:
https://github.com/danielperna84/custom_homematic#homematicip_localforce_device_availability
and
https://github.com/danielperna84/custom_homematic?tab=readme-ov-file#available-blueprints
and
https://github.com/danielperna84/hahomematic/discussions/566

**Comment by udochrist** [comment]
Thanks. I‘ve setup from a blueprint and the device gets re-activated.
